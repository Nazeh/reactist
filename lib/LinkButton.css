.reactist.link_button {
  text-decoration: none;
}
.reactist.link_button:active {
  color: #3F82EF;
}
.reactist.link_button:visited {
  color: #3F82EF;
}
.reactist.link_button.disabled {
  color: #808080;
}


/*# sourceMappingURL=LinkButton.map*/