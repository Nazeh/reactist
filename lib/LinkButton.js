!function(t,e){"object"==typeof exports&&"object"==typeof module?module.exports=e(require("prop-types"),require("react"),require("classnames")):"function"==typeof define&&define.amd?define("reactist",["prop-types","react","classnames"],e):"object"==typeof exports?exports.reactist=e(require("prop-types"),require("react"),require("classnames")):t.reactist=e(t["prop-types"],t.React,t.classnames)}(window,function(t,e,n){return function(t){var e={};function n(r){if(e[r])return e[r].exports;var o=e[r]={i:r,l:!1,exports:{}};return t[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=t,n.c=e,n.d=function(t,e,r){n.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:r})},n.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},n.t=function(t,e){if(1&e&&(t=n(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var o in t)n.d(r,o,function(e){return t[e]}.bind(null,o));return r},n.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return n.d(e,"a",e),e},n.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},n.p="",n(n.s=36)}({0:function(e,n){e.exports=t},1:function(t,n){t.exports=e},2:function(t,e){t.exports=n},3:function(t,e){t.exports=function(t){var e=[];return e.toString=function(){return this.map(function(e){var n=function(t,e){var n=t[1]||"",r=t[3];if(!r)return n;if(e&&"function"==typeof btoa){var o=(a=r,"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(a))))+" */"),i=r.sources.map(function(t){return"/*# sourceURL="+r.sourceRoot+t+" */"});return[n].concat(i).concat([o]).join("\n")}var a;return[n].join("\n")}(e,t);return e[2]?"@media "+e[2]+"{"+n+"}":n}).join("")},e.i=function(t,n){"string"==typeof t&&(t=[[null,t,""]]);for(var r={},o=0;o<this.length;o++){var i=this[o][0];"number"==typeof i&&(r[i]=!0)}for(o=0;o<t.length;o++){var a=t[o];"number"==typeof a[0]&&r[a[0]]||(n&&!a[2]?a[2]=n:n&&(a[2]="("+a[2]+") and ("+n+")"),e.push(a))}},e}},36:function(t,e,n){"use strict";n.r(e);n(37);var r=n(1),o=n.n(r),i=n(0),a=n.n(i),s=n(2),c=n.n(s);function u(t){return(u="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(t)}function f(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}function l(t){return(l=Object.setPrototypeOf?Object.getPrototypeOf:function(t){return t.__proto__||Object.getPrototypeOf(t)})(t)}function p(t){if(void 0===t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return t}function d(t,e){return(d=Object.setPrototypeOf||function(t,e){return t.__proto__=e,t})(t,e)}var b=function(t){function e(){var t,n,r,o,i,a,s;!function(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}(this,e);for(var c=arguments.length,f=new Array(c),d=0;d<c;d++)f[d]=arguments[d];return r=this,o=(t=l(e)).call.apply(t,[this].concat(f)),n=!o||"object"!==u(o)&&"function"!=typeof o?p(r):o,i=p(n),s=function(t){t.preventDefault(),!n.props.disabled&&n.props.onClick&&n.props.onClick()},(a="_handleClick")in i?Object.defineProperty(i,a,{value:s,enumerable:!0,configurable:!0,writable:!0}):i[a]=s,n}var n,r,i;return function(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function");t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,writable:!0,configurable:!0}}),e&&d(t,e)}(e,o.a.Component),n=e,(r=[{key:"render",value:function(){var t=c()("reactist link_button",{disabled:this.props.disabled},this.props.className);return o.a.createElement("a",{className:t,href:"",onClick:this._handleClick},this.props.name)}}])&&f(n.prototype,r),i&&f(n,i),e}();b.displayName="LinkButton",b.defaultProps={disabled:!1},b.propTypes={name:a.a.string.isRequired,onClick:a.a.func,disabled:a.a.bool,className:a.a.string},e.default=b},37:function(t,e,n){var r=n(38);"string"==typeof r&&(r=[[t.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(t.exports=r.locals)},38:function(t,e,n){(t.exports=n(3)(!1)).push([t.i,".reactist.link_button {\n  text-decoration: none;\n}\n.reactist.link_button:active {\n  color: #3F82EF;\n}\n.reactist.link_button:visited {\n  color: #3F82EF;\n}\n.reactist.link_button.disabled {\n  color: #808080;\n}\n",""])},4:function(t,e,n){var r,o,i={},a=(r=function(){return window&&document&&document.all&&!window.atob},function(){return void 0===o&&(o=r.apply(this,arguments)),o}),s=function(t){var e={};return function(t,n){if("function"==typeof t)return t();if(void 0===e[t]){var r=function(t,e){return e?e.querySelector(t):document.querySelector(t)}.call(this,t,n);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(t){r=null}e[t]=r}return e[t]}}(),c=null,u=0,f=[],l=n(5);function p(t,e){for(var n=0;n<t.length;n++){var r=t[n],o=i[r.id];if(o){o.refs++;for(var a=0;a<o.parts.length;a++)o.parts[a](r.parts[a]);for(;a<r.parts.length;a++)o.parts.push(m(r.parts[a],e))}else{var s=[];for(a=0;a<r.parts.length;a++)s.push(m(r.parts[a],e));i[r.id]={id:r.id,refs:1,parts:s}}}}function d(t,e){for(var n=[],r={},o=0;o<t.length;o++){var i=t[o],a=e.base?i[0]+e.base:i[0],s={css:i[1],media:i[2],sourceMap:i[3]};r[a]?r[a].parts.push(s):n.push(r[a]={id:a,parts:[s]})}return n}function b(t,e){var n=s(t.insertInto);if(!n)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=f[f.length-1];if("top"===t.insertAt)r?r.nextSibling?n.insertBefore(e,r.nextSibling):n.appendChild(e):n.insertBefore(e,n.firstChild),f.push(e);else if("bottom"===t.insertAt)n.appendChild(e);else{if("object"!=typeof t.insertAt||!t.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=s(t.insertAt.before,n);n.insertBefore(e,o)}}function y(t){if(null===t.parentNode)return!1;t.parentNode.removeChild(t);var e=f.indexOf(t);e>=0&&f.splice(e,1)}function v(t){var e=document.createElement("style");if(void 0===t.attrs.type&&(t.attrs.type="text/css"),void 0===t.attrs.nonce){var r=function(){0;return n.nc}();r&&(t.attrs.nonce=r)}return h(e,t.attrs),b(t,e),e}function h(t,e){Object.keys(e).forEach(function(n){t.setAttribute(n,e[n])})}function m(t,e){var n,r,o,i;if(e.transform&&t.css){if(!(i="function"==typeof e.transform?e.transform(t.css):e.transform.default(t.css)))return function(){};t.css=i}if(e.singleton){var a=u++;n=c||(c=v(e)),r=j.bind(null,n,a,!1),o=j.bind(null,n,a,!0)}else t.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(n=function(t){var e=document.createElement("link");return void 0===t.attrs.type&&(t.attrs.type="text/css"),t.attrs.rel="stylesheet",h(e,t.attrs),b(t,e),e}(e),r=function(t,e,n){var r=n.css,o=n.sourceMap,i=void 0===e.convertToAbsoluteUrls&&o;(e.convertToAbsoluteUrls||i)&&(r=l(r));o&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var a=new Blob([r],{type:"text/css"}),s=t.href;t.href=URL.createObjectURL(a),s&&URL.revokeObjectURL(s)}.bind(null,n,e),o=function(){y(n),n.href&&URL.revokeObjectURL(n.href)}):(n=v(e),r=function(t,e){var n=e.css,r=e.media;r&&t.setAttribute("media",r);if(t.styleSheet)t.styleSheet.cssText=n;else{for(;t.firstChild;)t.removeChild(t.firstChild);t.appendChild(document.createTextNode(n))}}.bind(null,n),o=function(){y(n)});return r(t),function(e){if(e){if(e.css===t.css&&e.media===t.media&&e.sourceMap===t.sourceMap)return;r(t=e)}else o()}}t.exports=function(t,e){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(e=e||{}).attrs="object"==typeof e.attrs?e.attrs:{},e.singleton||"boolean"==typeof e.singleton||(e.singleton=a()),e.insertInto||(e.insertInto="head"),e.insertAt||(e.insertAt="bottom");var n=d(t,e);return p(n,e),function(t){for(var r=[],o=0;o<n.length;o++){var a=n[o];(s=i[a.id]).refs--,r.push(s)}t&&p(d(t,e),e);for(o=0;o<r.length;o++){var s;if(0===(s=r[o]).refs){for(var c=0;c<s.parts.length;c++)s.parts[c]();delete i[s.id]}}}};var g,w=(g=[],function(t,e){return g[t]=e,g.filter(Boolean).join("\n")});function j(t,e,n,r){var o=n?"":r.css;if(t.styleSheet)t.styleSheet.cssText=w(e,o);else{var i=document.createTextNode(o),a=t.childNodes;a[e]&&t.removeChild(a[e]),a.length?t.insertBefore(i,a[e]):t.appendChild(i)}}},5:function(t,e){t.exports=function(t){var e="undefined"!=typeof window&&window.location;if(!e)throw new Error("fixUrls requires window.location");if(!t||"string"!=typeof t)return t;var n=e.protocol+"//"+e.host,r=n+e.pathname.replace(/\/[^\/]*$/,"/");return t.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(t,e){var o,i=e.trim().replace(/^"(.*)"$/,function(t,e){return e}).replace(/^'(.*)'$/,function(t,e){return e});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i)?t:(o=0===i.indexOf("//")?i:0===i.indexOf("/")?n+i:r+i.replace(/^\.\//,""),"url("+JSON.stringify(o)+")")})}}})});
//# sourceMappingURL=LinkButton.map