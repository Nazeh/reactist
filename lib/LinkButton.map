{"version":3,"sources":["webpack://reactist/webpack/universalModuleDefinition","webpack://reactist/webpack/bootstrap","webpack://reactist/external \"prop-types\"","webpack://reactist/external {\"root\":\"React\",\"commonjs2\":\"react\",\"commonjs\":\"react\",\"amd\":\"react\"}","webpack://reactist/external \"classnames\"","webpack://reactist/./src/components/LinkButton.jsx"],"names":["root","factory","exports","module","require","define","amd","window","__WEBPACK_EXTERNAL_MODULE__0__","__WEBPACK_EXTERNAL_MODULE__1__","__WEBPACK_EXTERNAL_MODULE__2__","installedModules","__webpack_require__","moduleId","i","l","modules","call","m","c","d","name","getter","o","Object","defineProperty","enumerable","get","r","Symbol","toStringTag","value","t","mode","__esModule","ns","create","key","bind","n","object","property","prototype","hasOwnProperty","p","s","LinkButton","event","preventDefault","_this","props","disabled","onClick","React","Component","className","classNames","this","react__WEBPACK_IMPORTED_MODULE_1___default","a","createElement","href","_handleClick","displayName","defaultProps","propTypes","PropTypes","string","isRequired","func","bool"],"mappings":"CAAA,SAAAA,EAAAC,GACA,iBAAAC,SAAA,iBAAAC,OACAA,OAAAD,QAAAD,EAAAG,QAAA,cAAAA,QAAA,SAAAA,QAAA,eACA,mBAAAC,eAAAC,IACAD,OAAA,+CAAAJ,GACA,iBAAAC,QACAA,QAAA,SAAAD,EAAAG,QAAA,cAAAA,QAAA,SAAAA,QAAA,eAEAJ,EAAA,SAAAC,EAAAD,EAAA,cAAAA,EAAA,MAAAA,EAAA,YARA,CASCO,OAAA,SAAAC,EAAAC,EAAAC,GACD,mBCTA,IAAAC,KAGA,SAAAC,EAAAC,GAGA,GAAAF,EAAAE,GACA,OAAAF,EAAAE,GAAAX,QAGA,IAAAC,EAAAQ,EAAAE,IACAC,EAAAD,EACAE,GAAA,EACAb,YAUA,OANAc,EAAAH,GAAAI,KAAAd,EAAAD,QAAAC,IAAAD,QAAAU,GAGAT,EAAAY,GAAA,EAGAZ,EAAAD,QA0DA,OArDAU,EAAAM,EAAAF,EAGAJ,EAAAO,EAAAR,EAGAC,EAAAQ,EAAA,SAAAlB,EAAAmB,EAAAC,GACAV,EAAAW,EAAArB,EAAAmB,IACAG,OAAAC,eAAAvB,EAAAmB,GAA0CK,YAAA,EAAAC,IAAAL,KAK1CV,EAAAgB,EAAA,SAAA1B,GACA,oBAAA2B,eAAAC,aACAN,OAAAC,eAAAvB,EAAA2B,OAAAC,aAAwDC,MAAA,WAExDP,OAAAC,eAAAvB,EAAA,cAAiD6B,OAAA,KAQjDnB,EAAAoB,EAAA,SAAAD,EAAAE,GAEA,GADA,EAAAA,IAAAF,EAAAnB,EAAAmB,IACA,EAAAE,EAAA,OAAAF,EACA,KAAAE,GAAA,iBAAAF,QAAAG,WAAA,OAAAH,EACA,IAAAI,EAAAX,OAAAY,OAAA,MAGA,GAFAxB,EAAAgB,EAAAO,GACAX,OAAAC,eAAAU,EAAA,WAAyCT,YAAA,EAAAK,UACzC,EAAAE,GAAA,iBAAAF,EAAA,QAAAM,KAAAN,EAAAnB,EAAAQ,EAAAe,EAAAE,EAAA,SAAAA,GAAgH,OAAAN,EAAAM,IAAqBC,KAAA,KAAAD,IACrI,OAAAF,GAIAvB,EAAA2B,EAAA,SAAApC,GACA,IAAAmB,EAAAnB,KAAA+B,WACA,WAA2B,OAAA/B,EAAA,SAC3B,WAAiC,OAAAA,GAEjC,OADAS,EAAAQ,EAAAE,EAAA,IAAAA,GACAA,GAIAV,EAAAW,EAAA,SAAAiB,EAAAC,GAAsD,OAAAjB,OAAAkB,UAAAC,eAAA1B,KAAAuB,EAAAC,IAGtD7B,EAAAgC,EAAA,GAIAhC,IAAAiC,EAAA,sBClFA1C,EAAAD,QAAAM,mBCAAL,EAAAD,QAAAO,mBCAAN,EAAAD,QAAAQ,20BCMMoC,idACa,SAAAC,GACXA,EAAMC,kBACDC,EAAKC,MAAMC,UAAYF,EAAKC,MAAME,SACnCH,EAAKC,MAAME,0PAJEC,IAAMC,4FASvB,IAAMC,EAAYC,IACd,wBAEIL,SAAUM,KAAKP,MAAMC,UAEzBM,KAAKP,MAAMK,WAEf,OACIG,EAAAC,EAAAC,cAAA,KAAGL,UAAWA,EAAWM,KAAK,GAAGT,QAASK,KAAKK,cAC1CL,KAAKP,MAAM7B,eAK5ByB,EAAWiB,YAAc,aACzBjB,EAAWkB,cACPb,UAAU,GAEdL,EAAWmB,WAEP5C,KAAM6C,IAAUC,OAAOC,WAEvBhB,QAASc,IAAUG,KAEnBlB,SAAUe,IAAUI,KAEpBf,UAAWW,IAAUC,QAGVrB","file":"LinkButton.js","sourcesContent":["(function webpackUniversalModuleDefinition(root, factory) {\n\tif(typeof exports === 'object' && typeof module === 'object')\n\t\tmodule.exports = factory(require(\"prop-types\"), require(\"react\"), require(\"classnames\"));\n\telse if(typeof define === 'function' && define.amd)\n\t\tdefine(\"reactist\", [\"prop-types\", \"react\", \"classnames\"], factory);\n\telse if(typeof exports === 'object')\n\t\texports[\"reactist\"] = factory(require(\"prop-types\"), require(\"react\"), require(\"classnames\"));\n\telse\n\t\troot[\"reactist\"] = factory(root[\"prop-types\"], root[\"React\"], root[\"classnames\"]);\n})(window, function(__WEBPACK_EXTERNAL_MODULE__0__, __WEBPACK_EXTERNAL_MODULE__1__, __WEBPACK_EXTERNAL_MODULE__2__) {\nreturn "," \t// The module cache\n \tvar installedModules = {};\n\n \t// The require function\n \tfunction __webpack_require__(moduleId) {\n\n \t\t// Check if module is in cache\n \t\tif(installedModules[moduleId]) {\n \t\t\treturn installedModules[moduleId].exports;\n \t\t}\n \t\t// Create a new module (and put it into the cache)\n \t\tvar module = installedModules[moduleId] = {\n \t\t\ti: moduleId,\n \t\t\tl: false,\n \t\t\texports: {}\n \t\t};\n\n \t\t// Execute the module function\n \t\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\n\n \t\t// Flag the module as loaded\n \t\tmodule.l = true;\n\n \t\t// Return the exports of the module\n \t\treturn module.exports;\n \t}\n\n\n \t// expose the modules object (__webpack_modules__)\n \t__webpack_require__.m = modules;\n\n \t// expose the module cache\n \t__webpack_require__.c = installedModules;\n\n \t// define getter function for harmony exports\n \t__webpack_require__.d = function(exports, name, getter) {\n \t\tif(!__webpack_require__.o(exports, name)) {\n \t\t\tObject.defineProperty(exports, name, { enumerable: true, get: getter });\n \t\t}\n \t};\n\n \t// define __esModule on exports\n \t__webpack_require__.r = function(exports) {\n \t\tif(typeof Symbol !== 'undefined' && Symbol.toStringTag) {\n \t\t\tObject.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });\n \t\t}\n \t\tObject.defineProperty(exports, '__esModule', { value: true });\n \t};\n\n \t// create a fake namespace object\n \t// mode & 1: value is a module id, require it\n \t// mode & 2: merge all properties of value into the ns\n \t// mode & 4: return value when already ns object\n \t// mode & 8|1: behave like require\n \t__webpack_require__.t = function(value, mode) {\n \t\tif(mode & 1) value = __webpack_require__(value);\n \t\tif(mode & 8) return value;\n \t\tif((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;\n \t\tvar ns = Object.create(null);\n \t\t__webpack_require__.r(ns);\n \t\tObject.defineProperty(ns, 'default', { enumerable: true, value: value });\n \t\tif(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));\n \t\treturn ns;\n \t};\n\n \t// getDefaultExport function for compatibility with non-harmony modules\n \t__webpack_require__.n = function(module) {\n \t\tvar getter = module && module.__esModule ?\n \t\t\tfunction getDefault() { return module['default']; } :\n \t\t\tfunction getModuleExports() { return module; };\n \t\t__webpack_require__.d(getter, 'a', getter);\n \t\treturn getter;\n \t};\n\n \t// Object.prototype.hasOwnProperty.call\n \t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\n\n \t// __webpack_public_path__\n \t__webpack_require__.p = \"\";\n\n\n \t// Load entry module and return exports\n \treturn __webpack_require__(__webpack_require__.s = 21);\n","module.exports = __WEBPACK_EXTERNAL_MODULE__0__;","module.exports = __WEBPACK_EXTERNAL_MODULE__1__;","module.exports = __WEBPACK_EXTERNAL_MODULE__2__;","import './styles/link_button.less'\n\nimport React from 'react'\nimport PropTypes from 'prop-types'\nimport classNames from 'classnames'\n\nclass LinkButton extends React.Component {\n    _handleClick = event => {\n        event.preventDefault()\n        if (!this.props.disabled && this.props.onClick) {\n            this.props.onClick()\n        }\n    }\n\n    render() {\n        const className = classNames(\n            'reactist link_button',\n            {\n                disabled: this.props.disabled\n            },\n            this.props.className\n        )\n        return (\n            <a className={className} href=\"\" onClick={this._handleClick}>\n                {this.props.name}\n            </a>\n        )\n    }\n}\nLinkButton.displayName = 'LinkButton'\nLinkButton.defaultProps = {\n    disabled: false\n}\nLinkButton.propTypes = {\n    /** Name of the link button that should be displayed. */\n    name: PropTypes.string.isRequired,\n    /** Function that should be called when link button is clicked. No parameters are passed. */\n    onClick: PropTypes.func,\n    /** Disabled link buttons cannot be clicked. */\n    disabled: PropTypes.bool,\n    /** Additional css class applied to the link button. */\n    className: PropTypes.string\n}\n\nexport default LinkButton\n"],"sourceRoot":""}