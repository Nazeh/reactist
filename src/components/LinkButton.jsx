import './styles/link_button.less'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class LinkButton extends React.Component {
    _handleClick = event => {
        event.preventDefault()
        if (!this.props.disabled && this.props.onClick) {
            this.props.onClick()
        }
    }

    render() {
        const className = classNames(
            'reactist_link_button',
            {
                disabled: this.props.disabled
            },
            this.props.className
        )
        return (
            <a className={className} href="" onClick={this._handleClick}>
                {this.props.name}
            </a>
        )
    }
}
LinkButton.displayName = 'LinkButton'
LinkButton.defaultProps = {
    disabled: false
}
LinkButton.propTypes = {
    /** Name of the link button that should be displayed. */
    name: PropTypes.string.isRequired,
    /** Function that should be called when link button is clicked. No parameters are passed. */
    onClick: PropTypes.func,
    /** Disabled link buttons cannot be clicked. */
    disabled: PropTypes.bool,
    /** Additional css class applied to the link button. */
    className: PropTypes.string
}

export default LinkButton
