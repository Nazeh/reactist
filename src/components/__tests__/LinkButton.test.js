import React from 'react'
import { shallow } from 'enzyme'

import LinkButton from '../LinkButton'

describe('LinkButton', () => {
    it('renders without crashing', () => {
        const linkButton = shallow(<LinkButton name="link" />)
        expect(linkButton).toMatchSnapshot()
    })

    it('appends additionally supplied classnames', () => {
        const linkButton = shallow(
            <LinkButton name="link" className="extra-brand-style" />
        )
        expect(linkButton).toMatchSnapshot()
    })

    it('calls onClick handler when supplied and prevents default behaviour of link', () => {
        const clickSpy = jest.fn()
        const defaultPreventionSpy = jest.fn()
        const linkButton = shallow(
            <LinkButton name="link" onClick={clickSpy} />
        )

        linkButton.simulate('click', { preventDefault: defaultPreventionSpy })
        expect(clickSpy).toHaveBeenCalled()
        expect(defaultPreventionSpy).toHaveBeenCalled()
    })

    it('does not call onClick handler when disabled but prevents default behaviour of link', () => {
        const clickSpy = jest.fn()
        const defaultPreventionSpy = jest.fn()
        const linkButton = shallow(
            <LinkButton name="link" disabled onClick={clickSpy} />
        )

        linkButton.simulate('click', { preventDefault: defaultPreventionSpy })
        expect(clickSpy).not.toHaveBeenCalled()
        expect(defaultPreventionSpy).toHaveBeenCalled()
    })
})
