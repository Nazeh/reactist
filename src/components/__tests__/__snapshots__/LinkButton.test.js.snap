// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton appends additionally supplied classnames 1`] = `
<a
  className="reactist_link_button extra-brand-style"
  href=""
  onClick={[Function]}
>
  link
</a>
`;

exports[`LinkButton renders without crashing 1`] = `
<a
  className="reactist_link_button"
  href=""
  onClick={[Function]}
>
  link
</a>
`;
